library/cwm/src/lib/cwm/dialog.rb

```diff
@@ -0,0 +1,97 @@
+require "yast"
+
+Yast.import "CWM"
+
+module CWM
+  # An OOP API and the pieces missing from {YastClass::CWM#show Yast::CWM.show}:
+  # - creating and closing a wizard dialog
+  # - Back/Abort/Next buttons
+  #
+  # @see UI::Dialog
+  # @see CWM::AbstractWidget
+  class Dialog
+    include Yast::Logger
+    include Yast::I18n
+    include Yast::UIShortcuts
+
+    # @return [String,nil] Set a title, or keep the existing title
+    abstract_method :title
+
+    # @return [CWM::WidgetTerm]
+    abstract_method :contents
+
+    # A shortcut for `.new(*args).run`
+    def self.run(*args)
+      new(*args).run
+    end
+
+    # The entry point.
+    # Will open (and close) a wizard dialog unless one already exists.
+    # @return [Symbol]
+    def run
+      if should_open_dialog?
+        wizard_create_dialog { run_assuming_open }
+      else
+        run_assuming_open
+      end
+    end
+
+    def wizard_create_dialog(&block)
+      Yast::Wizard.CreateDialog
+      block.call
+    ensure
+      Yast::Wizard.CloseDialog
+    end
+
+    def run_assuming_open
+      Yast::CWM.show(
+        contents,
+        caption:        title,
+        back_button:    replace_true(back_button, Yast::Label.BackButton),
+        abort_button:   replace_true(abort_button, Yast::Label.AbortButton),
+        next_button:    replace_true(next_button, Yast::Label.NextButton),
+        skip_store_for: skip_store_for
+      )
+    end
+
+    def should_open_dialog?
+      !Yast::Wizard.IsWizardDialog
+    end
+
+    # The :back button
+    # @return [String,true,nil] button label,
+    #   `true` to use the default label, or `nil` to omit the button
+    def back_button
+      true
+    end
+
+    # The :abort button
+    # @return [String,true,nil] button label,
+    #   `true` to use the default label, or `nil` to omit the button
+    def abort_button
+      true
+    end
+
+    # The :next button
+    # @return [String,true,nil] button label,
+    #   `true` to use the default label, or `nil` to omit the button
+    def next_button
+      true
+    end
+
+    # @return [Array<Symbol>]
```

Comment:
well, in fact I do not test it. But from past I know that any object stop during handle and convert to symbol looks happen only after test for skip for store https://github.com/yast/yast-yast2/blob/15350bbdd7bda71215caa24b114d027cb6b47e80/library/cwm/src/modules/CWM.rb#L874